File: module-4/mcp-dispatchkit-core/src/tools/get-vehicle-status.ts
import { z } from "zod";
import { vehicles } from "../data/vehicles.js";

export const getVehicleStatusInputShape = {
  vehicleId: z.string(),
};

const getVehicleStatusInput = z.object(getVehicleStatusInputShape);

export const getVehicleStatusDescription = "Get vehicle status.";

export async function getVehicleStatusHandler(
  args: z.infer<typeof getVehicleStatusInput>
) {
  const vehicle = vehicles.find((v) => v.id === args.vehicleId);

  return {
    content: [
      {
        type: "text" as const,
        text: vehicle ? JSON.stringify(vehicle, null, 2) : "",
      },
    ],
  };
}

File: module-4/python/mcp-dispatchkit-core/src/tools/get_vehicle_status.py
import json
from dataclasses import asdict

from ..data.vehicles import VEHICLES


def get_vehicle_status(vehicle_id: str) -> str:
    """Get vehicle status."""
    vehicle = next((v for v in VEHICLES if v.id == vehicle_id), None)

    if vehicle is None:
        return ""
    return json.dumps(asdict(vehicle), indent=2)
